libs/dal/src/shared/interceptors/invalidate-cache.interceptor.ts

```diff
@@ -8,22 +8,28 @@ export function InvalidateCache(storeKeyPrefix?: string) {
     descriptor.value = async function (...args: any[]) {
       if (!isStoreConnected(this.cacheService?.getStatus())) return await originalMethod.apply(this, args);
 
-      const query = args[0];
+      const res = await originalMethod.apply(this, args);
+
+      if (!res) {
+        return res;
+      }
+
+      const query = key === 'create' ? res : args[0];
 
       const cacheKey = buildKey(storeKeyPrefix ?? this.MongooseModel?.modelName, query);
 
       if (!cacheKey) {
-        return await originalMethod.apply(this, args);
+        return res;
       }
 
       try {
         this.cacheService.delByPattern(cacheKey);
+
+        return res;
       } catch (err) {
         // Logger.error(`An error has occurred when deleting "key: ${cacheKey}",`, 'InvalidateCache');
```

Comment:
For now let's atleast do a `console.error` here otherwise we are blind on what's going here

Reply:
Yes, I still did not manage to do the logging here, probably will need an instance of the logrepo here. wanted to consult about the logging strategy before doing so. I will add console.error at the moment.